subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformationRegistrationFactory.java

```diff
@@ -63,6 +67,8 @@ public DefaultTransformationRegistrationFactory(
         ClassLoaderHierarchyHasher classLoaderHierarchyHasher,
         TransformerInvoker transformerInvoker,
         ValueSnapshotter valueSnapshotter,
+        FileCollectionFactory fileCollectionFactory,
+        FileCollectionFingerprinterRegistry fileCollectionFingerprinterRegistry,
```

Comment:
I find it strange that we have to hold onto these things already when we are creating the registration. Can we pass them in some later time?

Reply:
We hold on to many things when we create the registration (basically everything), since we don't have a single place to pass in all of this at a later time.

I think we need to do something about this when the classpath normalization requires to have effect on artifact transform parameters/inputs as well, since when the transform is registered the classpath normalization strategy is not yet available.

Do you have an idea for a place where we could inject the fingerprinters later?

Reply:
Let's defer cleaning this up till https://github.com/gradle/build-cache/issues/1501.